### ISSUE TITLE
[YAML] Completion for referenced types should work even when '#' of json-pointer is not escaped

### ISSUE THREAD

**Issue by viatra2** [zalando/intellij-swagger#22]
Consider: 

```
swagger: 2.0
paths:
  /pets:
    get:
      responses:
        200:
          schema:
            $ref: #<caret>
definitions:
  Pets:
  Error:
```

Here, the `#` is recognised as a start of yaml comment, so no completion is currently suggested. 
If user explicitly escaped `'#` then both `#/definitions/Pets` and `#definitions/Error` are correctly suggested. 

However, the need for escaping of the `#` symbol is not obvious, so it would be nice if the same completion set would be suggested (and choosing one of them automatically escaped the reference). 

Note that in this case user may only type `sc<completion><enter>$<completion><enter>#<completion>` to finish with the correct response definition

**Comment by semonte** [comment]
@viatra2 I can take a look at this today

**Comment by semonte** [comment]
Fixed in dc3c03ed443e40820da3398924fb29c4d384e90f.
